src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -35,12 +35,18 @@ void ColumnVsValueTableScanImpl::_scan_non_reference_segment(
     return;
   }
 
-  // Select optimized or generic scanning implementation based on segment type
-  if (const auto* dictionary_segment = dynamic_cast<const BaseDictionarySegment*>(&segment)) {
-    _scan_dictionary_segment(*dictionary_segment, chunk_id, matches, position_filter);
-  } else {
-    _scan_generic_segment(segment, chunk_id, matches, position_filter);
-  }
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
```

Comment:
Why do you need to call resolve_data_and_segment_type? You aren't using type nor typed_segment.

Reply:
You're right, this was a remnant of a previous version of the code.